java/http/java-http-consumer/src/main/java/HttpConsumer.java

```diff
@@ -108,6 +135,19 @@ public void createConsumer() throws IOException, InterruptedException, URISyntax
         }
     }
 
+    public void processProperty(Properties props, String k, Map<String, Object> map) {
+        if (props.stringPropertyNames().contains(k)) {
+            String v = props.getProperty(k);
+            Object obj = v;
+            if (isStringBoolean(v)) {
+                obj = Boolean.parseBoolean(v);
+            } else if (isStringNumber(v)) {
+                obj = Integer.parseInt(v);
+            }
+            map.put(k, obj);
+        }
+    }
+
```

Comment:
I would change the `processProperty` method into `createConsumerConfig` method like this
```suggestion
  public void createConsumerConfig(Properties props) {
    Map<String, Object> consumerConfig = new HashMap<>();
    consumerConfig.put("name", "my-consumer");
    consumerConfig.put("format", "json");
    for (String k : commonProps) {
        if (props.stringPropertyNames().contains(k)) {
            String v = props.getProperty(k);
            Object obj = v;
            if (isStringBoolean(v)) {
                obj = Boolean.parseBoolean(v);
            } else if (isStringNumber(v)) {
                obj = Integer.parseInt(v);
            }
            consumerConfig.put(k, obj);
        }
        return consumerConfig;
  }
```

Reply:
It's not just about the `processProperty` as it is today but I see that the proposal from @kyguy includes the loop on the `commonProps` as well. I like the idea.

Reply:
Refactored. `  public Map<String, Object> createConsumerConfig(Properties props) ` Map return type, not void.